src/shared/dissect-image.c

```diff
@@ -2260,6 +2260,9 @@ int dissected_image_mount(
                                 if (r > 0)
                                         esp_path = "/efi";
                         }
+
+                        if (!esp_path)
+                                return -ENOENT; /* none of the possible directory is empty. */
```

Comment:
this is a bit too generic an error. find some other error code that is less likely to clash with errors the kernel might give us